generic_expression_patterns_modules/process.py

```diff
@@ -572,12 +574,50 @@ def process_raw_compendium_recount2(
     normalize_compendium(mapped_filename, normalized_filename, scaler_filename)
 
 
+def format_pseudomonas_pathway_DB(pathway_DB_filename, local_dir, out_filename):
+    """
+    This function reads in pseudomonas pathway data from 
+    `pathway_DB_filename` and formats and outputs it
+    to `output_filename` in order to be
+    used in GSEA_analysis.R
+
+    Note: Currently this function is specifically
+    customized to expect pathway_DB_filename = 
+    "https://raw.githubusercontent.com/greenelab/adage/master/Node_interpretation/pseudomonas_KEGG_terms.txt"
+
+    """
+    # Read in pathway data
+    pa_pathway_DB = pd.read_csv(pathway_DB_filename, sep="\t", header=None)
+
+    # Drop extra column
+    pa_pathway_DB.drop(columns=[1], inplace=True)
+
+    # Make genes tab-separated
+    pa_pathway_DB[2] = pa_pathway_DB[2].str.split(";").str.join("\t")
+
+    # Need to temporarily write data to file in order
```

Comment:
Not sure I quite follow why you need to write this file twice - is the extra `\` part of the tabs that you're adding? I don't see any forward slashes in the original file, but maybe I'm missing something.

Reply:
I was trying to write a csv file where each row was a different length, because each pathway has a different number of genes. When pandas tries to write the df which has the list of genes as a string, it tries to "escape" the commas. I'm trying to purposely unescape them with this line.

Reply:
Got it. But shouldn't the lists of genes be tab-separated strings (since you did a string join in line 596 above)?

I guess it's not a big deal if you're only doing this once for a relatively short file, but it feels to me like there's probably a way to do this without writing the file twice. Maybe just make a note to look into this further if it's slow in the future?

Reply:
Sorry I mis-typed. That should have said "tabs". This was what I found: https://github.com/pandas-dev/pandas/issues/16298

So at least I don't think that is currently a way to do it